packages/cozy-stack-client/src/CozyStackClient.js

```diff
@@ -205,8 +205,10 @@ class CozyStackClient {
   }
 
   async fetchJSONWithCurrentToken(method, path, body, options = {}) {
-    const headers = (options.headers = options.headers || {})
-
+    //Since we modify the object later by adding in some case a
+    //content-type, let's clone this object to scope the modification
+    const clonedOptions = { ...options }
+    const headers = (clonedOptions.headers = clonedOptions.headers || {})
```

Comment:
Nit: If there are nested options, we can still have a bug since the rest of the method could change stuff (thinking "oh the var is called 'cloned', there is no problem to change any value inside"). It would be even safer to deepClone the object.

Reply:
Nice catch! Changed to use lodash.cloneDeep()